### PR TITLE
docs(server-communication): e2e tests added

### PR DESCRIPTION
**Pull request by Dotneteer**
Here are the E2E tests added to the server-communication chapter.

### CODE REVIEW

File: public/docs/_examples/server-communication/e2e-spec.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      var heroTags = heroListComp.all(by.tagName('li'));
+      expect(heroTags.count()).toBe(_initialHeroCount);
+    });
+    
+    it('should not add hero with empty name', function () {
+      var myTohComp = element(by.tagName('my-toh'));
+      expect(myTohComp).toBeDefined('<my-toh> must exist');
+      var addButton = myTohComp.element(by.tagName('button'));
+      expect(addButton).toBeDefined('"Add Hero" button must be defined');
+      addButton.click(function() {
```
Comment (wardbell) [inline public/docs/_examples/server-communication/e2e-spec.js]: The following kind of test, which you use repeatedly, does not work 
```
addButton.click(function() {...})
```
The function is never executed; the test always passes.

The `click` method returns a promise. The correct syntax invokes the function within the `then` success callback:
```
addButton.click().then(function() { ... })
```

Comment (Dotneteer) [inline public/docs/_examples/server-communication/e2e-spec.js]: Fixed. Changed all occurrences (2) to the proper pattern.

**Comment by Dotneteer** [comment]
I’m the author of these changes.

**Comment by wardbell** [comment]
@Dotneteer  Your claims about the empty list in wiki tests are not true and your test does not confirm it; you do not do as you say here:

>With the 'testForRefreshedResult()' method we check that the result list is immediately deleted after
      sending a key to the search field,

 I tried to come up with a test that would confirm/deny it and I can't. The old list is always populated until replaced by the new.

Therefore, I removed your comments with what seem to me to be false claims.

I agree that it is not easy to test the difference between the two wiki requests. The challenge is that we cannot know if wikipedia will respond in under 300ms.

I don't think this is worth testing right now.

**Pull request closed without merging**